src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/TempPE/TempPEBuildManager.cs

```diff
@@ -0,0 +1,550 @@
+            }
+
+            // Apply our new value
+            AppliedValue = new ProjectVersionedValue<DesignTimeInputsItem>(new DesignTimeInputsItem
+            {
+                Inputs = newDesignTimeInputs,
+                SharedInputs = newSharedDesignTimeInputs,
+                // We always need an output path, so if it hasn't changed we just reuse the previous value
+                OutputPath = value.OutputPath ?? previousValue.OutputPath,
+                Cookies = newCookies,
+                TaskSchedulers = newSchedulers
+            }, value.DataSourceVersions);
+
+            // Project properties changes cause all PEs to be dirty and recompile if this isn't the first update
+            if (value.HasProjectPropertyChanges)
+            {
+                foreach (string item in newDesignTimeInputs)
+                {
+                    await FireTempPEDirtyAsync(item, value.ShouldCompile);
+                }
+            }
+            else
+            {
+                // Individual inputs dirty their PEs and possibly recompile
+                foreach (string item in addedDesignTimeInputs)
+                {
+                    await FireTempPEDirtyAsync(item, value.ShouldCompile);
+                }
+            }
+
+            // Shared items cause all TempPEs to be dirty, but don't recompile, to match legacy behaviour
+            if (hasRemovedDesignTimeSharedInputs || hasAddedDesignTimeSharedInputs)
+            {
+                // adding or removing shared design time inputs dirties things but doesn't recompile
+                foreach (string item in newDesignTimeInputs)
+                {
+                    // We don't want to fire again if we already fired above and compiled
+                    if (!addedDesignTimeInputs.Contains(item))
+                    {
+                        await FireTempPEDirtyAsync(item, false);
+                    }
+                }
+            }
+
+            foreach (string item in removedDesignTimeInputs)
+            {
+                _buildManager.Value.OnDesignTimeOutputDeleted(item);
+            }
+
+            bool TryGetValueIfUnused<T>(string item, ImmutableDictionary<string, T>.Builder source, ImmutableHashSet<string>.Builder otherSources, out T result)
+            {
+                // return the value from the source, but only if it doesn't appear in other sources
+                return source.TryGetValue(item, out result) && !otherSources.Contains(item);
+            }
+
+        }
+
+        protected virtual ITaskDelayScheduler CreateTaskScheduler()
+        {
+            return new TaskDelayScheduler(s_compilationWaitTime, UnconfiguredProjectServices.ThreadingService, DisposalToken);
+        }
+
+        protected override async Task DisposeCoreAsync(bool initialized)
+        {
+            if (initialized)
+            {
+                DisposeTaskSchedulers();
+            }
+            await base.DisposeCoreAsync(initialized);
```

Comment:
return.

Reply:
🤦‍♂️ 

This probably explains the analyzer complaining, I thought it was a false positive.

Reply:
No analyzer will still complain.